backend/de.metas.handlingunits.base/src/main/java/de/metas/picking/service/impl/HU2PackingItemsAllocator.java

```diff
@@ -103,6 +113,8 @@
 	private IHUContext _huContext = null;
 	private Quantity _qtyToPackRemaining = null;
 
+	@NonNull final Map<HuId, PickedHuAndQty> changedPickedHus = new HashMap<>();
```

Comment:
wondering why not recording all transactions, not only those which split off the HU.
In this way I think it would be easier to handle and easier to read.